pkg/utils/utils.go

```diff
@@ -237,3 +270,37 @@ func GetUIODeviceFile(dev string) (devFile string, err error) {
 
 	return
 }
+
+// GetNetNames returns host net interface names as string for a PCI device from its pci address
+func GetNetNames(pciAddr string) ([]string, error) {
+	var names []string
+	netDir := filepath.Join(sysBusPci, pciAddr, "net")
+	if _, err := os.Lstat(netDir); err != nil {
+		return nil, fmt.Errorf("GetNetName(): no net directory under pci device %s: %q", pciAddr, err)
+	}
+
+	fInfos, err := ioutil.ReadDir(netDir)
+	if err != nil {
+		return nil, fmt.Errorf("GetNetName(): failed to read net directory %s: %q", netDir, err)
+	}
+
+	names = make([]string, 0)
+	for _, f := range fInfos {
+		names = append(names, f.Name())
+	}
+
+	return names, nil
+}
+
+// GetDriverName returns current driver attached to a pci device from its pci address
+func GetDriverName(pciAddr string) (string, error) {
+	driverLink := filepath.Join(sysBusPci, pciAddr, "driver")
+	driverInfo, err := os.Readlink(driverLink)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return pciAddr, nil
```

Comment:
is it intended behaviour to return PCI address as the driver name in case the file doesn't exist? or is it leftover from copy-pasted `GetPfAddr` function?

Reply:
We should return error for all errors during driver name look up. The driver name is required to make decision on creating resource pool. I will fix this. Thanks